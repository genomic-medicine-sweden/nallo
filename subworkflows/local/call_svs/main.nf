include { ADD_FOUND_IN_TAG    } from '../../../modules/local/add_found_in_tag/main'
include { BCFTOOLS_MERGE      } from '../../../modules/nf-core/bcftools/merge/main'
include { BCFTOOLS_QUERY      } from '../../../modules/nf-core/bcftools/query/main'
include { BCFTOOLS_REHEADER   } from '../../../modules/nf-core/bcftools/reheader/main'
include { CREATE_SAMPLES_FILE } from '../../../modules/local/create_samples_file/main'
include { SEVERUS             } from '../../../modules/nf-core/severus/main'
include { SNIFFLES            } from '../../../modules/nf-core/sniffles/main'

workflow CALL_SVS {

    take:
    ch_bam_bai        // channel [mandatory]: [ val(meta), path(bam), path(bai) ]
    ch_fasta          // channel [mandatory]: [ val(meta), path(fasta) ]
    ch_fai            // channel [mandatory]: [ val(meta), path(fai) ]
    sv_caller         //     val [mandatory]: Which caller to use
    ch_tandem_repeats // channel  [optional]: [ val(meta), path(bed) ]

    main:
    ch_versions     = Channel.empty()

    // Call SVs
    if (sv_caller == "severus") {

        SEVERUS (
            ch_bam_bai.map { meta, bam, bai -> [ meta, bam, bai, [], [], [] ] },
            ch_tandem_repeats
        )
        ch_versions = ch_versions.mix(SEVERUS.out.versions)

        SEVERUS.out.all_vcf
            .set { ch_vcf }

    } else if (sv_caller == "sniffles") {

        SNIFFLES (
            ch_bam_bai,
            ch_fasta,
            ch_tandem_repeats,
            true,
            false
        )
        ch_versions = ch_versions.mix(SNIFFLES.out.versions)

        SNIFFLES.out.vcf
            .set { ch_vcf }
    }

    // Annotate with FOUND_IN tag
    ADD_FOUND_IN_TAG (
        ch_vcf.map { meta, vcf -> [ meta, vcf, [] ] },
        sv_caller
    )
    ch_versions = ch_versions.mix(ADD_FOUND_IN_TAG.out.versions)

    // Get the sample name from the VCF
    // For Sniffles this is hardcoded as SAMPLE and for Severus it's based on the filename
    BCFTOOLS_QUERY (
        ADD_FOUND_IN_TAG.out.vcf.join(ADD_FOUND_IN_TAG.out.csi),
        [],
        [],
        []
    )
    ch_versions = ch_versions.mix(BCFTOOLS_QUERY.out.versions)

    // Creates a "vcf_sample_name meta.id" file for bcftools reheader
    CREATE_SAMPLES_FILE ( BCFTOOLS_QUERY.out.output )
    ch_versions = ch_versions.mix(CREATE_SAMPLES_FILE.out.versions)

    ADD_FOUND_IN_TAG.out.vcf
        .join( CREATE_SAMPLES_FILE.out.samples )
        .map { meta, vcf, samples -> [ meta, vcf, [], samples ] }
        .set { ch_bcftools_reheader_in }

    // Give meta.id as sample name in the VCF
    BCFTOOLS_REHEADER ( ch_bcftools_reheader_in, [[],[]] )
    ch_versions = ch_versions.mix(BCFTOOLS_REHEADER.out.versions)

    BCFTOOLS_REHEADER.out.vcf
        .join(BCFTOOLS_REHEADER.out.index)
        .map { meta, vcf, tbi -> [ [ 'id': meta.project ], vcf, tbi ] }
        .groupTuple()
        .set { ch_bcftools_merge_in }

    // Merge the files with new sample names
    BCFTOOLS_MERGE ( ch_bcftools_merge_in, ch_fasta, ch_fai, [[],[]] )
    ch_versions = ch_versions.mix(BCFTOOLS_MERGE.out.versions)

    emit:
    ch_sv_calls_vcf     = BCFTOOLS_REHEADER.out.vcf   // channel: [ val(meta), path(vcf) ]
    ch_sv_calls_tbi     = BCFTOOLS_REHEADER.out.index // channel: [ val(meta), path(tbi) ]
    ch_multisample_vcf  = BCFTOOLS_MERGE.out.vcf      // channel: [ val(meta), path(vcf) ]
    ch_multisample_tbi  = BCFTOOLS_MERGE.out.index    // channel: [ val(meta), path(tbi) ]
    versions            = ch_versions                 // channel: [ path(versions.yml) ]
}

