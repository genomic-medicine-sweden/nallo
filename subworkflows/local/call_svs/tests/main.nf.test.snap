{
    "1 sample - [bam, bai], fasta, fai, [] - severus": {
        "content": [
            "1294e2ab468789a0bd56e8bcaf578a4",
            "1294e2ab468789a0bd56e8bcaf578a4",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:35:49.943901515"
    },
    "1 sample - [bam, bai], fasta, fai, [] - sniffles": {
        "content": [
            "7cd0730159250c8945e2007da414651e",
            "5db7be988de18b4f14b532043adb367f",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:34:57.588634749"
    },
    "1 sample - [bam, bai], fasta, fai, bed - severus": {
        "content": [
            "d64a5438fe54c0239613976f8826897b",
            "d64a5438fe54c0239613976f8826897b",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:36:09.577048048"
    },
    "1 sample - [bam, bai], fasta, fai, [] - severus -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                    "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
                ],
                "ch_multisample_tbi": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_multisample_vcf": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "ch_sv_calls_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_sv_calls_vcf": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                    "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T13:00:41.241344463"
    },
    "2 samples - [bam, bai], fasta, fai, [] - sniffles -stub": {
        "content": [
            "[test_1_reheader.vcf.gz, test_2_reheader.vcf.gz]",
            [
                [
                    {
                        "id": "test_1",
                        "single_end": false,
                        "project": "project"
                    },
                    "test_1_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "id": "test_2",
                        "single_end": false,
                        "project": "project"
                    },
                    "test_2_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "project"
                    },
                    "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                ]
            ],
            [
                [
                    {
                        "id": "project"
                    },
                    "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T13:00:27.793986704"
    },
    "1 sample - [bam, bai], fasta, fai, bed - severus -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                    "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
                ],
                "ch_multisample_tbi": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_multisample_vcf": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "ch_sv_calls_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_sv_calls_vcf": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                    "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T13:00:54.535163836"
    },
    "1 sample - [bam, bai], fasta, fai, [] - sniffles -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
                ],
                "ch_multisample_tbi": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_multisample_vcf": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "ch_sv_calls_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_sv_calls_vcf": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:59:59.327540921"
    },
    "2 samples - [bam, bai], fasta, fai, [] - severus -stub": {
        "content": [
            "[test_1_reheader.vcf.gz, test_2_reheader.vcf.gz]",
            [
                [
                    {
                        "id": "test_1",
                        "single_end": false,
                        "project": "project"
                    },
                    "test_1_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    {
                        "id": "test_2",
                        "single_end": false,
                        "project": "project"
                    },
                    "test_2_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                [
                    {
                        "id": "project"
                    },
                    "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                ]
            ],
            [
                [
                    {
                        "id": "project"
                    },
                    "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T13:01:08.741446388"
    },
    "1 sample - [bam, bai], fasta, fai, bed - sniffles": {
        "content": [
            "3cb4ba0bf05ab17ef02c4b51e70e4e6e",
            "c7e259bedb8c0e4742bf1a5722ac2973",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:35:13.337758686"
    },
    "2 samples - [bam, bai], fasta, fai, [] - sniffles": {
        "content": [
            "[test_1_reheader.vcf.gz, test_2_reheader.vcf.gz]",
            "86c9d09c440d7d60876ae6b4db8a86ab",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:35:33.058751753"
    },
    "1 sample - [bam, bai], fasta, fai, bed - sniffles -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
                ],
                "ch_multisample_tbi": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_multisample_vcf": [
                    [
                        {
                            "id": "project"
                        },
                        "project.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "ch_sv_calls_tbi": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "ch_sv_calls_vcf": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "project": "project"
                        },
                        "test_reheader.vcf.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                    "versions.yml:md5,572bfd042014333af194207291d12b95",
                    "versions.yml:md5,589fd8607c7131f786d565deb2846f0c",
                    "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                    "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                    "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T13:00:13.310974915"
    },
    "2 samples - [bam, bai], fasta, fai, [] - severus": {
        "content": [
            "[test_1_reheader.vcf.gz, test_2_reheader.vcf.gz]",
            "37b9bd8c61a49f47e07db24a911f3cce",
            [
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,110bc7290a8aa06aea1c1fd2bad26324",
                "versions.yml:md5,572bfd042014333af194207291d12b95",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,7090d52a64ab671fdb51c6af429628f4",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,a41129f62453d49bc1979c1a20551d7c",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,ad7a1d3e0b9738f34f6f4cf8c7a5a7ff",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f",
                "versions.yml:md5,dab46ec91d22db3eee78879773b88f1f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-09-17T12:37:01.825339869"
    }
}